java/performance/src/test/java/org/apache/arrow/adapter/jdbc/JdbcAdapterBenchmarks.java

```diff
@@ -0,0 +1,331 @@
+
+    @Setup(Level.Trial)
+    public void prepareState() throws Exception {
+      allocator = new RootAllocator(Integer.MAX_VALUE);
+      config = new JdbcToArrowConfigBuilder().setAllocator(allocator).setTargetBatchSize(1024).build();
+      Class.forName(DRIVER);
+      conn = DriverManager.getConnection(URL);
+
+      try (Statement stmt = conn.createStatement()) {
+        stmt.executeUpdate(CREATE_STATEMENT);
+      }
+
+      for (int i = 0; i < VALUE_COUNT; i++) {
+        // Insert data
+        try (PreparedStatement stmt = conn.prepareStatement(INSERT_STATEMENT)) {
+
+          stmt.setInt(1, i);
+          stmt.setLong(2, i);
+          stmt.setString(3, "test" + i);
+          stmt.setBoolean(4, i % 2 == 0);
+          stmt.executeUpdate();
+        }
+      }
+    }
+
+    @Setup(Level.Invocation)
+    public void prepareInvoke() throws Exception {
+      statement = conn.createStatement();
+      resultSet = statement.executeQuery(QUERY);
+    }
+
+    @TearDown(Level.Invocation)
+    public void tearDownInvoke() throws Exception {
+      resultSet.close();
+      statement.close();
+    }
+
+    @TearDown(Level.Trial)
+    public void tearDownState() throws Exception {
+      try (Statement stmt = conn.createStatement()) {
+        stmt.executeUpdate(DROP_STATEMENT);
+      }
+      allocator.close();
+    }
+  }
+
+  /**
+   * State object for the consume benchmark.
+   */
+  @State(Scope.Benchmark)
+  public static class ConsumeState {
+
+    private static final boolean NULLABLE = false;
+
+    private Connection conn = null;
+
+    private ResultSet resultSet = null;
+
+    private BaseAllocator allocator;
+
+    private Statement statement;
+
+    private IntVector intVector;
+
+    private BigIntVector longVector;
+
+    private JdbcConsumer<IntVector> intConsumer;
+
+    private JdbcConsumer<BigIntVector> longConsumer;
+
+    private JdbcToArrowConfig config;
+
+    @Setup(Level.Trial)
+    public void prepare() throws Exception {
+      allocator = new RootAllocator(Integer.MAX_VALUE);
+      config = new JdbcToArrowConfigBuilder().setAllocator(allocator).setTargetBatchSize(1024).build();
+
+      Class.forName(DRIVER);
+      conn = DriverManager.getConnection(URL);
+      try (Statement stmt = conn.createStatement()) {
+        stmt.executeUpdate(CREATE_STATEMENT);
+      }
+
+      for (int i = 0; i < VALUE_COUNT; i++) {
+        // Insert data
+        try (PreparedStatement stmt = conn.prepareStatement(INSERT_STATEMENT)) {
+
+          stmt.setInt(1, i);
+          stmt.setLong(2, i);
+          stmt.setString(3, "test" + i);
+          stmt.setBoolean(4, i % 2 == 0);
+          stmt.executeUpdate();
+        }
+      }
+
+      statement = conn.createStatement();
+      resultSet = statement.executeQuery(QUERY);
+      resultSet.next();
+
+      intVector = new IntVector("", allocator);
+      intVector.allocateNew(VALUE_COUNT);
+      intConsumer = IntConsumer.createConsumer(intVector, 1, NULLABLE);
+
```

Comment:
hmm, does it change performance if you call IntConsumer.createConsumer(intVector, 1, /*nullable*/=true) here?

Reply:
I have tried this. There is no performance change. 
I think for JIT, a static final is almost the same as a constant literal.

Reply:
I'm asking to create a NullableConsumer for int so the JVM is forced to load both leaf classes.  Is that what you meant?

Reply:
I see your point. Thanks. Will try it later.

Reply:
I have tried to create both versions (nullable & non-nullable) of consumers for both columns, and there is no observable performance degradation:

Benchmark                               Mode  Cnt    Score   Error  Units
JdbcAdapterBenchmarks.consumeBenchmark  avgt    5  117.875 ± 0.469  us/op